Documentation updates for DataPrepper 2.8 (#7135)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/processors/write_json.md

---
layout: default
title: write_json
parent: Processors
grand_parent: Pipelines
nav_order: 56
---

# write_json

> The `write_json` processor converts an object in the event into a JSON string. You can customize the processor to choose the source and target field names.

Reviewer: ```suggestion
The `write_json` processor converts an object in an event into a JSON string. You can customize the processor to choose the source and target field names.
```

> You can use the `write_json` processor to write an object in the event as JSON string. You can customize the `write_json` processor to pick the source and target field names

Reviewer 2: ```suggestion
The `write_json` processor converts an object in the event into a JSON string. You can customize the processor to choose the source and target field names.
```

Reviewer 2: To automatically choose the source and target?

| Option | Description | Example |
| :--- | :--- | :--- |

> | source | The message field to be parsed. Mandatory field name to identify the source object in the event. | If `source` is `"message"`, `{"message": {"key1":"value1", "key2":{"key3":"value3"}}`, write_json processor, generates `{"message": "{\"key1\":\"value`\", \"key2\":"{\"key3\":\"value3\"}"}"`

Reviewer 2: ```suggestion
| source | Mandatory field that specifies the name of the field in the event containing the message or object to be parsed. | If `source` is `"message"`, `{"message": {"key1":"value1", "key2":{"key3":"value3"}}`, the `write_json` processor, generates `{"message": "{\"key1\":\"value`\", \"key2\":"{\"key3\":\"value3\"}"}"`
```

> | source | Mandatory field that specifies the name of the field in the event containing the message or object to be parsed. | If `source` is `"message"`, `{"message": {"key1":"value1", "key2":{"key3":"value3"}}`, the `write_json` processor, generates `{"message": "{\"key1\":\"value`\", \"key2\":"{\"key3\":\"value3\"}"}"`

Reviewer: I'm not following the example here. Are the tic marks/commas in the right places, and should there be a "then" in there somewhere?

> | target | target field name under which the JSON string is stored. This is Optional field. If not specified, `source` is used as the target.

Reviewer 2: ```suggestion
| target | An optional field that specifies the name of the field where the resulting JSON string should be stored. If `target` is not specified, then the `source` field is used.
```

> | target | An optional field that specifies the name of the field where the resulting JSON string should be stored. If `target` is not specified, then the `source` field is used.

Reviewer: ```suggestion
| target | An optional field that specifies the name of the field in which the resulting JSON string should be stored. If `target` is not specified, then the `source` field is used.
```
